src/sqlfluff/core/templaters/placeholder.py

```diff
@@ -41,6 +41,10 @@
     "dollar": regex.compile(
         r"(?<![:\w\x5c])\${?(?P<param_name>[\w_]+)}?", regex.UNICODE
     ),
+    # e.g. WHERE bla = $name or WHERE bla = ${name} (DbUp compatible)
+    "dollar_surround": regex.compile(
+        r"(?<![:\w\x5c])\$(?P<param_name>[\w_]+)\$?", regex.UNICODE
```

Comment:
The documentation suggests that a `-` is valid inside the dollar quotes (`\w` should include `_`). Also, that second `$` I don't think is supposed to be optional.
```suggestion
        r"(?<![:\w\x5c])\$(?P<param_name>[-\w]+)\$", regex.UNICODE
```